pkg/minikube/service/service_test.go

```diff
@@ -712,50 +713,49 @@ preferences: {}
 users:
 - name: minikube
 `
-	var tests = []struct {
-		description    string
-		kubeconfigPath string
-		config         string
-		err            bool
+	tests := []struct {
+		description string
+		config      string
+		err         bool
 	}{
 		{
-			description:    "ok",
-			kubeconfigPath: "/tmp/kube_config",
-			config:         mockK8sConfig,
-			err:            false,
+			description: "ok",
+			config:      mockK8sConfig,
+			err:         false,
 		},
 		{
-			description:    "empty config",
-			kubeconfigPath: "/tmp/kube_config",
-			config:         "",
-			err:            true,
+			description: "empty config",
+			config:      "",
+			err:         true,
 		},
 		{
-			description:    "broken config",
-			kubeconfigPath: "/tmp/kube_config",
-			config:         "this**is&&not: yaml::valid: file",
-			err:            true,
+			description: "broken config",
+			config:      "this**is&&not: yaml::valid: file",
+			err:         true,
 		},
 	}
 
 	for _, test := range tests {
 		t.Run(test.description, func(t *testing.T) {
-			mockK8sConfigByte := []byte(test.config)
-			mockK8sConfigPath := test.kubeconfigPath
-			err := os.WriteFile(mockK8sConfigPath, mockK8sConfigByte, 0644)
-			defer os.Remove(mockK8sConfigPath)
-			if err != nil {
-				t.Fatalf("Unexpected error when writing to file %v. Error: %v", test.kubeconfigPath, err)
+			// Each sub-test gets its own temp dir & file for isolation and portability.
+			tmpDir := t.TempDir()
+			mockK8sConfigPath := filepath.Join(tmpDir, "kube_config")
+
+			// Only write the file if we have any content; an empty string still creates a 0-byte file,
+			// which is what we want to simulate "empty config".
+			if err := os.WriteFile(mockK8sConfigPath, []byte(test.config), 0644); err != nil {
```

Comment:
There is no reason to create the kubeconfig world readable. This is the default kubeconfig permissions on macOS:

```
% ls -lh ~/.kube/config
-rw-------  1 nir  staff   106B Oct 10 11:37 /Users/nir/.kube/config
```

It will be better to use real world permissions. This will break the code if it has bad assumptions about being able to read the config from another user in the same group.